hardware/raspberrypi/bootmodes/gpio.md

```diff
@@ -24,7 +24,7 @@ Once enabled, the GPIOs are allocated to boot modes as follows:
 
 The various boot modes are attempted in the numerical order of the GPIO pins, i.e. SD0, then SD1, then NAND and so on.
 
-SD0 is the Broadcom SD card / MMC interface. When the boot ROM within the SoC runs, it always connects SD0 to the built-in SD card slot. On Compute Modules with an eMMC device, SD0 is connected to that; on the Compute Module Lite SD0 is available on the edge connector and connects to the SD card slot in the CMIO carrier board. SD1 is the Arasan SD card / MMC interface which is also capable of SDIO. All Raspberry Pi models with built-in wifi use SD1 to connect to the wifi chip via SDIO.
+SD0 is the Broadcom SD card / MMC interface. When the boot ROM within the SoC runs, it always connects SD0 to the built-in microSD card slot. On Compute Modules with an eMMC device, SD0 is connected to that; on the Compute Module Lite SD0 is available on the edge connector and connects to the microSD card slot in the CMIO carrier board. SD1 is the Arasan SD card / MMC interface which is also capable of SDIO. All Raspberry Pi models with built-in wifi use SD1 to connect to the wifi chip via SDIO.
```

Comment:
Perhaps the first "microSD" here should be "SD / microSD"? (because the original Pis had full-size SD slots)

Reply:
That was deliberate - I was assuming GPIO boot mode is only available on the 2837. But that has not been confirmed - paging @ghollingworth 😉

Reply:
Ahhh, I hadn't considered that. But even _without_ GPIO boot mode, "When the boot ROM within the SoC runs, it always connects SD0 to the built-in SD card slot." is still true for the original Raspberry Pi models :wink:

Reply:
If GPIO boot mode is available on the old Pi's then I will change the wording, otherwise it is correct as-is.

Reply:
GPIO boot mode is only available on 3B and 3B+

Reply:
Thanks @ghollingworth. So just to confirm - the 2837 checks to see which model if Pi it is running, and only allows GPIO boot mode if it is in a 3B or 3B+, so on the 3A+ and the 2837-based compute modules setting program_gpio_bootmode=n does nothing?

Reply:
No - the ROM knows nothing about the board it is fitted to. I think what @ghollingworth meant is that GPIO boot mode is new for 2837, so not available on earlier Pis that don't have one.

Reply:
I wouldn't have thought so - otherwise why mention "the GPIOs in bank 2 are only available on the Compute Module" ? :man_shrugging: 
I wonder if Gordon meant "GPIO boot mode is only available on the BCM2837A0 and BCM2837B0" ?

Reply:
Thanks - I've added 3A+ and CM3 and 3+. There's also the 2837-based Pi 2B just to confuse things, but I'm not sure that's really worth documenting.

Reply:
Just need @ghollingworth to confirm that GPIO boot mode is only available on **all** 2837-based Pi's, then this PR is ready for CE. 😀